src/ffi/ffi_bind_utils.ts

```diff
@@ -1,20 +1,24 @@
 import { ptr, FFIType } from 'bun:ffi'
 
-function arrayArg(arg, type) {
+function arrayArg(arg: number | number[], type: FFIType) {
+  if (typeof arg === 'number') arg = [arg]
+
   if (arg.length == 0) {
     return [0, 0]
   }
-  if (arg.constructor === Number) {
-    arg = [arg]
-  }
+
   let array: Int32Array | BigInt64Array
-  if (type === FFIType.i32) {
-    array = new Int32Array(arg)
-  } else if (type === FFIType.i64) {
-    array = new BigInt64Array(arg.map((x) => BigInt(x)))
-  } else {
-    throw 'unhandled type for array argument'
+  switch (type) {
```

Comment:
interesting! did you get a change to benchmark the change?  this can definitely end up on the hot path for models with transposes/reshapes in them

Reply:
In testing against the same setup with multiple conditionals, the performance difference was negligible, but from my understanding switch case tends to generally slightly outperform conditionals (and the perf gains increase as the # of conditions to check increases).


"As it turns out, the switch statement is faster in most cases when compared to if-else, but significantly faster only when the number of conditions is large. The primary difference in performance between the two is that the incremental cost of an additional condition is larger for if-else than it is for switch." Source: https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html

Reply:
So as we handle more FFIType's, the change to switch case should be advantageous (especially considering how frequently this function is called).